app/controllers/needs_controller.rb

```diff
@@ -46,23 +46,27 @@ def create
       render(:new)
     end
   end
-
+  
   def edit
     @need = policy_scope(Need).find(params[:id])
-
+    
     authorize @need
   end
-
+  
   def update
     @need = policy_scope(Need).find(params[:id])
     @need.assign_attributes(permitted_attributes(@need))
-
+    
     authorize @need
-
-    if @need.save
-      Services::Notifications::Needs.new(@need, :update).notify
-      redirect_to(@need)
+    if params[:need][:children_attributes].select { |_, v| v[:_destroy] == ''}.keys.any?
```

Comment:
@mireie did you investigate why the model-level validation isn't preventing all the children from being removed? I presume there is something wrong with it, but I'm not sure what it is.

This fix does indeed resolve the symptom I experienced on staging, but I think a deeper fix should be sought out. Happy to look into it with you sometime.

Reply:
I [did look into this](https://github.com/OfficeMomsandDads/scheduler/pull/446#issuecomment-962105870) and was not able to figure out why. I'd be happy to review.

Reply:
I still haven't figured out how to verify the model-level validation so I instead updated the React component to hide the delete button if there is only one child and added a check to `destroyChild` to only execute if there is more than one child.